ssh-uuid.sh

```diff
@@ -390,9 +390,13 @@ function get_rand_port_num {
 
 function check_tool {
 	local tool
-	tool=$(which "$1")
+	local tool_not_found
+	tool=$(which "$1") || tool_not_found=1
+	if [[ ! -z "$tool_not_found" ]]; then
```

Comment:
Actually, I was running some tests now and I found that the `which` command produces exit status code 1 (failure) if:
* Either the argument does not exist in PATH,
* Or the argument exists but does not have executable permissions.

Which means that the `-x` test is redundant and the function could be overall simplified as follows:
```bash
function check_tool {
	if [ ! $(which "$1") ]; then
		quit "'$1' not found in PATH. Is it installed?"
	fi
}
```
What do you think?

Reply:
thanks for your comments. I didn't know that `which` also checked if it was executable. I agree with the simplification. I updated the MR with that.

What version of shellcheck do you use? I ask because with the one available in debian 11 I get many more issues than those on the rest of the code. Debian 11 ships with shellcheck v0.5

Reply:
I have shellcheck v0.8.0:
```
$ shellcheck --version
ShellCheck - shell script analysis tool
version: 0.8.0
license: GNU General Public License, version 3
website: https://www.shellcheck.net
```
It was installed through Homebrew (macOS), and it looks like the latest version from the developer:
* https://github.com/koalaman/shellcheck/releases
* https://formulae.brew.sh/formula/shellcheck

Reply:
FYI @cbecker, I've made a further change/simplification to the function in bf6029c9823adbfe9fa7dfca8df6466371d26a9a and now it looks like:
```bash
function check_tool {
	which "$1" &>/dev/null || quit "'$1' not found in PATH. Is it installed?"
}
```
When I proposed `[ ! $(which "$1") ]`, I was mistaken about what it meant (I should have known better!). I was thinking it was testing the exit status code of `which`, but it was actually testing the stdout of `which`, and ignoring its exit status code. If `jq` is installed, `which jq` prints e.g. `/usr/local/bin/jq` and the test expression became:
```
    if [ ! /usr/local/bin/jq ]; then
```
And if `jq` is not installed, `which jq` prints nothing and the expression became:
```
    if [ ! ]; then
```
It turns out that it works, but I was not comfortable with it as I had meant to test the exit status code of `which`, which is most easily achieved through the `||` operator in a single line - in idiomatic shell script form. :-)

I have added tag v0.2.2 for this change plus two other unrelated commits (README typos and test verbosity).
Thanks again!